src/app/pages/admin/private/shop-details/shop-details.component.ts

```diff
@@ -298,4 +301,32 @@ export class ShopDetailsComponent extends BaseComponent {
   acceptTermsAndConditions() {
     this.termsAndCondition = !this.termsAndCondition;
   }
+
+  getShopDetails() {
+    this.shopService.getShopDetails(this.shopId)
+      .then(response => this.initializeForm(response['data']['shop']))
+      .catch(error => {
+        this.router.navigate([APP_ROUTES.DASHBOARD])
+      })
+  }
+
+  updateShopActiveStatus() {
+    this.isShopActive = !this.isShopActive;
+    let requestBody = {}
+    requestBody['status'] = this.formStatus == 'ACTIVE' ? 'INACTIVE' : 'ACTIVE'
+    this.updateShopDetails(requestBody, {resetStatus: true})
+  }
+
+  updateShopDetails(requestBody, options = {}) {
+    this.shopService.updateShopDetails(this.shopId, requestBody)
+      .then(response => {
+        this.initializeForm(response['data']['shop']);
+      })
+      .catch(error => {
+        // TODO: navigate to list of shops
+        if (error['status'] == 404) this.router.navigateByUrl(APP_ROUTES.DASHBOARD);
+        if (options['resetStatus'] && options['resetStatus'] == true) this.isShopActive = !this.isShopActive;
```

Comment:
`if (options['resetStatus']) this.isShopActive = !this.isShopActive;`
Should be fine?